hipogram/posts/views.py

```diff
@@ -99,6 +97,30 @@ def update_post(request, id):
         return redirect("users:login")
 
 
+def like_post(request, post_id):
+    like, created = Like.objects.get_or_create(user=request.user, post_id=post_id)
+
+    if not created:
+        like.delete()
+
+    return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+
+def rate_post(request, post_id):
+
+    if request.method == 'POST':
+        form = RatePostForm(request.POST)
+
+    # if not form.is_valid():
+        # do something
+
+        rate, created = Rate.objects.get_or_create(user=request.user, post_id=post_id)
+        rate.value = form.data['value']
```

Comment:
I'm not sure if `form.data['value']` is available before running `form.is_valid()`. If this works, it's okay, no need to handle form validation here in our case.